impress-org/givewp
# Correct revenue data from imported donations to have minor amount

Resolves #5406 

## Description

There was an issue where donations being imported did not calulate the correct minor amount, so what was being stored in the `give_revenue` table was incorrect. For example:

$125 imported
Was stored as 125
Which equals $1.25

The issue had to do with the fact that the `Money` class was attempting to use currency settings to determine the number of decimal places for the currency. This is a bad idea decimals in the settings should only reflect decimals _displayed_ to the user. It does not affect how many decimals a currency actually supports — that's above the organization's power to determine.

This changes it so that the number of decimals is determined based on the actual currency's data, regardless of the organization's currency settings.

## Affects

Donation data stored in the `give_revenue` table either when a donation is made or imported.

## Pre-review Checklist




## Testing Instructions

1. Make a couple donations
2. Export the donations
3. Delete the donations from GiveWP
4. Import the donations
5. View the `give_revenue` table to make sure the amounts reflect the currency's smallest unit
